KEYCLOAK-7594 Upgrade to Wildfly 13, Upgrade JDG for cross-dc to 7.2

mposolda opened the pull request.

mingjliu9 commented on pom.xml:
```diff
@@ -705,8 +707,18 @@
             <dependency>
                 <groupId>org.jboss.aesh</groupId>
```
If my understanding is right, we no longer need this `org.jboss.aesh` dependency now.

mposolda replied:
@mingjliu9 It's unfortunately still needed as keycloak-admin-cli and keycloak-client-registration-cli are still using it. Both CLIs are packed in server distribution in directory KEYCLOAK_HOME/bin/client/ in the "fat" jars, which among other things, contain the classes of old aesh. I've created https://issues.jboss.org/browse/KEYCLOAK-7737 . 

Fortunately org.jboss.aesh is not needed in jboss modules, so IMO KEYCLOAK-7737 is not a blocker for this PR, but can be done later.

stianst commented on adapters/oidc/js/src/main/resources/keycloak.d.ts:
```diff
@@ -1,503 +1,503 @@
-		loadUserInfo(): KeycloakPromise<{}, void>;
-	}
-}
+/*
```
What's happening with this file? Some sort of encoding change?

mposolda replied:
Yes, it seems that my linux/github changed the line encodings of this file from LF to CRLF and seem it's doing it automatically. Will need to look at it to avoid this change...

stianst commented on distribution/feature-packs/server-feature-pack/src/main/resources/content/bin/migrate-domain-standalone.cli:
```diff
@@ -47,6 +47,7 @@ if (result == undefined) of /profile=$standaloneProfile/subsystem=infinispan/cac
   echo Updating authorization cache container..
   /profile=$standaloneProfile/subsystem=infinispan/cache-container=keycloak/local-cache=authorization/component=eviction/:write-attribute(name=strategy,value=LRU)
   /profile=$standaloneProfile/subsystem=infinispan/cache-container=keycloak/local-cache=authorization/component=eviction/:write-attribute(name=max-entries,value=100)
+  echo
```
What's with the empty echo statements here?

mposolda replied:
That's just for adding another line to the output of the CLI command. I've just tried without echo and the output doesn't look so nice. Also the pattern with empty echo is used in whole file and all the previous instructions.

mposolda replied:
Just to compare, here is the output file with empty echo command:
``
Removing eviction from keycloak users cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}

Removing eviction from keycloak authorization cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}

Removing eviction from keycloak keys cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}

``

And here the one without echo:
``
Removing eviction from keycloak users cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
Removing eviction from keycloak authorization cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
Removing eviction from keycloak keys cache and replacing with object-memory
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
{
    "outcome" => "success",
    "response-headers" => {"process-state" => "reload-required"}
}
``

My vote is, that the first one looks better

stianst commented on pom.xml:
```diff
@@ -44,16 +44,18 @@
-        <wildfly.build-tools.version>1.2.2.Final</wildfly.build-tools.version>
+        <wildfly.version>13.0.0.Final</wildfly.version>
+        <wildfly.build-tools.version>1.2.10.Final</wildfly.build-tools.version>
         <eap.version>7.1.3.GA-redhat-2</eap.version>
```
This needs to be changed to corresponding EAP CD build. That's 7.2.0.CD13-redhat-3. Also need to check that RH-SSO works as expected.

mposolda replied:
Ok, will look at it

stianst commented on pom.xml:
```diff
@@ -44,16 +44,18 @@
+        <wildfly.version>13.0.0.Final</wildfly.version>
+        <wildfly.build-tools.version>1.2.10.Final</wildfly.build-tools.version>
         <eap.version>7.1.3.GA-redhat-2</eap.version>
         <eap.build-tools.version>1.2.2.Final</eap.build-tools.version>
```
This also needs to be updated

stianst reviewed (changes requested):
Some changes needed, specifically changing EAP version for product profile so RH-SSO CD3 can be built. We also need to test that RH-SSO works.

We should probably run a full CI matrix on both Keycloak and RH-SSO before merging this as well.

mposolda commented:
@stianst Thanks for the review. Will try to look at keycloak.d.ts file and updating EAP version and running CI matrix for both Keycloak and RH-SSO.

vramik reviewed (commented):
ConsoleProtectionTest fails with Wildfly13. There might been some changes since WF11 .

hmlnarik reviewed (commented):
@mposolda Nice, thanks! I need to do play a bit with it but overall looks good to me.

mposolda committed:
KEYCLOAK-7594 Upgrade to Wildfly 13. Cross-DC: Upgrade to infinispan server 9.2.4 and JDG 7.2

Co-authored-by: Douglas Palmer <dpalmer@redhat.com>
Co-authored-by: stianst <stianst@gmail.com>
Co-authored-by: Hynek Mlnarik <hmlnarik@redhat.com>

mposolda commented:
@stianst Addressed your suggestions (despite empty echo statements in the migration scripts)

@hmlnarik I've figured that in the Keycloak distribution, we already removed the headers like "Server" and "X-Powered-By" . I think this was intentional to not reveal the things like version etc. So I've just removed it from this PR.

@vramik The ConsoleProtectionTest is still failing (sent mail to keycloak-team for decide how to move forward).

mposolda committed:
KEYCLOAK-8127 Added support for app-server-eap71. Make sure ConsoleProtectionTest is executed just for app-server-eap71

vramik commented on testsuite/integration-arquillian/servers/app-server/jboss/eap/src/main/java/org/keycloak/testsuite/arquillian/eap/container/EAPAppServerProvider.java:
```diff
@@ -40,6 +40,7 @@
     private final String startupTimeoutInSeconds;
 
     public EAPAppServerProvider() {
+        containerName = System.getProperty("app.server");
```
Wouldn't be better to also test that property is not null or empty? I know it is set by property in pom.xml but in case someone accidentally removes it it will start failing with NPE. See the lines below.

mposolda replied:
@vramik Sorry I missed this validation. Added: Validate.notNullOrEmpty(containerName, "app.server is not set.");

hmlnarik reviewed (commented):
@mposolda Thanks!
My concerns have been addressed, lgtm.

vramik reviewed (approved):
@mposolda thank you, lgtm
